### ISSUE TITLE
[Tenserflow]_[학습규제전략]

### ISSUE THREAD

**Issue by jangsik-park** [jangsik-park/Study#11]
# 학습규제전략 **(N413)**

- 신경망 훈련과정에서 _overfitting을 방지_하기 위한 방법입니다.
- 크게 Early stopping, weight decay, weight constraint, drop out이 있습니다.

## [1] Early stopping

- test set의 Error가 traing set의 Error와 차이가 벌어지기 시작하면 모델을 중단 시킵니다.

## [2] weight decay

- 학습을 진행하면서 특정 가중치 값들이 커지면 이를 조정하는 역할을 한다.
- L1 일반화 = 전체 Error에 가중치의 '절댓값'을 더해줘 ''가중치가 너무 크지 않은 방향으로 학습하게 됩닏다''
- L2 일반화 = 전체 Error에 가중치의 '제곱'을 더해줘 ''가중치가 너무 크지 않은 방향으로 학습하게 됩니다.''

# [3] weight constraint

- 가중치를 제거하거나 값을 규제하여 과적합을 방지한다.

# [4] drop out

- 데이터를 train 할 때 일부 노드를 사용하지 않고 학습을 하는 것이다.
- 분산을 줄여서 과적합을 방지하게 된다.
